build.sbt

```diff
@@ -1365,6 +1365,7 @@ lazy val `scio-smb`: Project = project
       "com.github.ben-manes.caffeine" % "caffeine" % caffeineVersion % Provided,
       "org.apache.avro" % "avro" % avroVersion % Provided,
       "org.apache.hadoop" % "hadoop-common" % hadoopVersion % Provided,
+      "org.apache.hadoop" % "hadoop-mapreduce-client-core" % hadoopVersion % Provided,
```

Comment:
without this, was getting a bunch of errors from my `ParquetUtils` class, i.e.:

```shell
[error] /workdir/scio/scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/ParquetUtils.java:63:1: cannot access org.apache.hadoop.mapreduce.lib.output.FileOutputFormat
[error]   class file for org.apache.hadoop.mapreduce.lib.output.FileOutputFormat not found
```

Reply:
The build complains about this scoping: it is not used by compilation.
The lib looks to be needed for runtime, so in our case this should be scoped to `Test` (we do not want to ship transitively)

I think we should revisit the build to mark `scio-parquet`, `scio-avro` or any IO module having an smb impl as provided instead of the storage format libs.

Reply:
See #5004

Reply:
hm, looks like that scope change [brings back](https://github.com/spotify/scio/actions/runs/6225615484/job/16896427165?pr=4980) the compilation errors:

```
/home/runner/work/scio/scio/scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/ParquetUtils.java:71:1: cannot access org.apache.hadoop.mapreduce.JobContext
```

maybe should be Runtime?

Reply:
added to the filter with comment. I think the tool is not smart enough to find this kind of transitive usage.